pkg/cloudprovider/aws/aws.go

```diff
@@ -73,12 +107,65 @@ func (f *ec2InstanceFilter) Matches(instance ec2.Instance) bool {
 }
 
 // goamzEC2 is an implementation of the EC2 interface, backed by goamz
-type GoamzEC2 struct {
-	ec2 *ec2.EC2
+type goamzEC2 struct {
+	auth aws.Auth
+	ec2  *ec2.EC2
+
+	mutex      sync.Mutex
+	elbClients map[string]*elb.ELB
+}
+
+func newGoamzEC2(auth aws.Auth, regionName string) (*goamzEC2, error) {
+	region, ok := aws.Regions[regionName]
+	if !ok {
+		return nil, fmt.Errorf("not a valid AWS region: %s", regionName)
+	}
+
+	self := &goamzEC2{}
+	self.ec2 = ec2.New(auth, region)
+	self.auth = auth
+	self.elbClients = make(map[string]*elb.ELB)
+	return self, nil
+}
+
+// Find the kubernetes vpc
+func (self *goamzEC2) ListVpcs(filterName string) ([]ec2.VPC, error) {
```

Comment:
Both here and in some other cases below (like getElbClient), the golang style guideline is for acronyms to be either all capital or all lower case, and in my experience the k8s code base does follow this guideline even if we don't follow everything (like some of the rules on comments). For example, this would be ListVPCs.
https://github.com/golang/go/wiki/CodeReviewComments#initialisms